Tasks/GooglePlayStatusUpdate/googleutil.ts

```diff
@@ -0,0 +1,169 @@
+    params?: PackageParams;
+}
+
+export function getJWT(key: ClientKey): JWT {
+    const GOOGLE_PLAY_SCOPES: string[] = ['https://www.googleapis.com/auth/androidpublisher'];
+    return new google.auth.JWT(key.client_email, null, key.private_key, GOOGLE_PLAY_SCOPES, null);
+}
+
+/**
+ * Uses the provided JWT client to request a new edit from the Play store and attach the edit id to all requests made this session
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @return {Promise} edit - A promise that will return result from inserting a new edit
+ *                          { id: string, expiryTimeSeconds: string }
+ */
+export async function getNewEdit(edits: any, globalParams: GlobalParams, packageName: string): Promise<Edit> {
+    tl.debug('Creating a new edit');
+    const requestParameters: PackageParams = {
+        packageName: packageName
+    };
+
+    tl.debug('Additional Parameters: ' + JSON.stringify(requestParameters));
+    const res = await edits.insert(requestParameters);
+    return res.data;
+}
+
+/**
+ * Gets information for the specified app and track
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @param {string} track - one of the values {"internal", "alpha", "beta", "production"}
+ * @returns {Promise} track - A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+export async function getTrack(edits: any, packageName: string, track: string): Promise<Track> {
+    tl.debug('Getting Track information');
+    const requestParameters: PackageParams = {
+        packageName: packageName,
+        track: track
+    };
+
+    tl.debug('Additional Parameters: ' + JSON.stringify(requestParameters));
+    const getTrack = await edits.tracks.get(requestParameters);
+    return getTrack.data;
+}
+
+/**
+ * Update a given release track with the given information
+ * Assumes authorized
+ * @param {string} packageName - unique android package name (com.android.etc)
+ * @param {string} track - one of the values {"internal", "alpha", "beta", "production"}
+ * @param {integer or [integers]} versionCode - version code returned from an apk call. will take either a number or a [number]
+ * @param {string} status - one of the values {"draft", "inProgress", "halted", "completed"}
+ * @param {double} userFraction - for rollouting out a release to a track.
+ * @param {releaseNotes} releaseNotes - optional release notes to be attached as part of the update
+ * @returns {Promise} track - A promise that will return result from updating a track
+ *                            { track: string, versionCodes: [integer], userFraction: double }
+ */
+export async function updateTrack(edits: any, packageName: string, track: string, versionCode: any, status: string, userFraction: number, releaseNotes?: ReleaseNotes[]): Promise<Track> {
```

Comment:
Could we use type `number | [number]` for `versionCode` parameter?

Reply:
`versionCodes`'s type in `androidpublisher_v3.Schema$TrackRelease` is `string[] | null`, if use use type `number | [number]` for `versionCode` parameter brings some problems, I'm working on it.

Reply:
Replace AndroidRelease with androidpublisher_v3.Schema$TrackRelease.
Could you please take a look again?
Thanks.